QLogger.cpp

```diff
@@ -1,266 +1,252 @@
-   QMutexLocker(&manager->mutex);
-
-   const auto logWriter = manager->getLogWriter(module);
-
-   if (logWriter && !logWriter->isStop() && logWriter->getLevel() <= level)
-   {
-      manager->writeAndDequeueMessages(module);
-      logWriter->write(module, message, level);
-   }
-   else if (!logWriter)
-      manager->queueMessage(
-          module,
-          { message, static_cast<int>(level), QDateTime::currentDateTime().toString("dd-MM-yyyy hh:mm:ss.zzz") });
+   QLoggerManager::getInstance()->enqueueMessage(module, level, message, function, file, line);
 }
 
 static const int QUEUE_LIMIT = 100;
 
-// QLoggerManager
-QLoggerManager *QLoggerManager::INSTANCE = nullptr;
-bool QLoggerManager::mIsStop = false;
-
 QLoggerManager::QLoggerManager()
-   : mutex(QMutex::Recursive)
+   : mMutex(QMutex::Recursive)
 {
-   QDir dir(QDir::currentPath());
-   dir.mkdir("logs");
+
 }
 
 QLoggerManager *QLoggerManager::getInstance()
 {
-   if (!INSTANCE)
-      INSTANCE = new QLoggerManager();
+   static QLoggerManager INSTANCE;
 
-   return INSTANCE;
+   return &INSTANCE;
 }
 
-QString QLoggerManager::levelToText(const LogLevel &level)
+bool QLoggerManager::addDestination(const QString &fileDest, const QString &module, LogLevel level,
+                                    const QString &fileFolderDestination, LogMode mode,
+                                    LogFileDisplay fileSuffixIfFull, LogMessageDisplays messageOptions, bool notify)
 {
-   switch (level)
-   {
-      case LogLevel::Trace:
-         return "Trace";
-      case LogLevel::Debug:
-         return "Debug";
-      case LogLevel::Info:
-         return "Info";
-      case LogLevel::Warning:
-         return "Warning";
-      case LogLevel::Error:
-         return "Error";
-      case LogLevel::Fatal:
-         return "Fatal";
-   }
+   QMutexLocker lock(&mMutex);
 
-   return QString();
-}
-
-bool QLoggerManager::addDestination(const QString &fileDest, const QString &module, LogLevel level)
-{
-   if (!moduleDest.contains(module))
+   if (!mModuleDest.contains(module))
    {
-      const auto log = new QLoggerWriter(fileDest, level);
+       const auto log = new QLoggerWriter((fileDest.isEmpty() ? mDefaultFileDestination : fileDest),
+                                          (level == LogLevel::Warning ? mDefaultLevel : level),
+                                          (fileFolderDestination.isEmpty() ? mDefaultFileDestinationFolder : QDir::fromNativeSeparators(fileFolderDestination)),
+                                          (mode == LogMode::OnlyFile ? mDefaultMode : mode),
+                                          (fileSuffixIfFull == LogFileDisplay::DateTime ? mDefaultFileSuffixIfFull : fileSuffixIfFull),
+                                          (messageOptions.testFlag(LogMessageDisplay::Default) ? mDefaultMessageOptions : messageOptions));
+      log->setMaxFileSize(mDefaultMaxFileSize);
       log->stop(mIsStop);
 
-      moduleDest.insert(module, log);
+      if (notify) {
```

Comment:
In the method that takes a QStringList of modules there is an additional check in here `if (!mIsStop)`. I think it's needed in here as well.

Reply:
You right!